src/Redux/Missions/missions.js

```diff
@@ -0,0 +1,58 @@
+const link = 'https://api.spacexdata.com/v3/missions';
+
+// REDUCERS
+const missionsReducer = (state = [], action) => {
+  switch (action.type) {
+    case 'loadMissions':
+      return action.playload;
+    case 'join': {
+      const id = action.playload;
+      const newState = state.map((mission) => {
+        if (mission.mission_id !== id) return mission;
+        return { ...mission, member: true };
+      });
+      return newState;
+    }
+    case 'leave': {
+      const key = action.playload;
+      const leaveMission = state.map((mission) => {
+        if (mission.mission_id !== key) return mission;
+        return { ...mission, member: false };
+      });
+      return leaveMission;
+    }
+    default:
+      return state;
+  }
+};
+
+// ACTIONS
+export const fetchMissions = () => async (dispatch) => {
+  const response = await fetch(link);
+  const data = await response.json();
+  const missions = [];
+  data.forEach((item) => {
+    const mission = {
+      mission_id: item.mission_id,
+      mission_name: item.mission_name,
+      description: item.description,
+      member: false,
+    };
+    missions.push(mission);
+  });
+  dispatch({
+    type: 'loadMissions',
+    playload: missions,
+  });
+};
```

Comment:
- [ ] [Optional] For a more professional approach all async actions put them in a separate .js module you can name it something like apiHelper.js and put it directly in the Redux directory.